mypy/server/update.py

```diff
@@ -246,20 +246,37 @@ def update(self,
             self.manager.log_fine_grained(
                 "update once: {} in {:.3f}s - {} left".format(
                     next_id, t1 - t0, len(changed_modules)))
-            if blocker:
+
+            if blocker_messages is not None:
                 self.blocking_error = (next_id, next_path)
                 self.stale = changed_modules
-                return messages
-
-        return messages
+                return blocker_messages
+
+            # It looks like we are done processing everything, so now
```

Comment:
Style nit: This function is getting uncomfortably long. Can you move the new code (or some of the old code) to another function to keep this more manageable?

Reply:
This is still an issue.

Reply:
Moved some of the old code out to a new function.